netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -593,13 +593,13 @@ protected boolean isGracefulShutdownComplete() {
    */
   private void createStream(CreateStreamCommand command, ChannelPromise promise)
           throws Exception {
-    if (lifecycleManager.getShutdownThrowable() != null) {
+    if (lifecycleManager.getShutdownStatus() != null) {
       command.stream().setNonExistent();
       // The connection is going away (it is really the GOAWAY case),
       // just terminate the stream now.
       command.stream().transportReportStatus(
           lifecycleManager.getShutdownStatus(), RpcProgress.MISCARRIED, true, new Metadata());
-      promise.setFailure(lifecycleManager.getShutdownThrowable());
+      promise.setFailure(lifecycleManager.getShutdownStatus().asException());
```

Comment:
I suggest using `InternalStatus.asRuntimeExceptionWithoutStacktrace()` here. The [receiver of this error](https://github.com/grpc/grpc-java/blob/21915575824a0f88fe468a87843977db0c9ba7fa/netty/src/main/java/io/grpc/netty/NettyClientStream.java#L158) is going to immediately convert it back into a Status. We don't use "without stacktrace" much, but we do use it a lot within NettyClientHandler already and when #10524 is complaining about allocating an exception, the vast majority of the cost (cpu and memory) of an exception is the stack trace. So I wouldn't suggest trying to hard to use it all over the place, but here it makes sense and furthers the goal of #10524.